src/helpers/cleanRelatedService.spec.js

```diff
@@ -0,0 +1,110 @@
+import { waitFor } from '@testing-library/react'
+
+import {
+  getHasManyItem,
+  removeHasManyItem
+} from 'cozy-client/dist/associations/HasMany'
+
+import { cleanRelatedContactsService } from './cleanRelatedContactsService'
+
+jest.mock('cozy-client', () => ({
+  Q: jest.fn()
+}))
+
+jest.mock('cozy-client/dist/associations/HasMany', () => ({
+  getHasManyItem: jest.fn(),
+  removeHasManyItem: jest.fn()
+}))
+
+const setup = ({
+  mockQueryAll = jest.fn(),
+  mockSaveAll = jest.fn(),
+  contacts = []
+} = {}) => {
+  const mockClient = {
+    queryAll: mockQueryAll.mockResolvedValue(contacts),
+    saveAll: mockSaveAll
+  }
+
+  return mockClient
+}
+
+describe('cleanRelatedContactsService', () => {
+  it('should update contacts with related contact deleted', async () => {
+    const mockSaveAll = jest.fn()
+    const contacts = [
+      {
+        _id: 'contact0',
+        relationships: {
+          related: { data: [{ _id: 'contactDeletedId' }] }
+        }
+      },
+      {
+        _id: 'contact1',
+        relationships: {
+          related: { data: [] }
+        }
+      }
+    ]
+    const client = setup({ mockSaveAll, contacts })
+
+    getHasManyItem.mockImplementation((contact, relation, id) => {
+      return contact.relationships[relation].data.find(item => item._id === id)
+    })
+
+    removeHasManyItem.mockImplementation((contact, relation, id) => {
+      return {
+        ...contact,
+        relationships: {
+          [relation]: {
+            data: contact.relationships.related.data.filter(
+              item => item._id !== id
+            )
+          }
+        }
+      }
+    })
```

Comment:
You had to mock these two functions ?

Reply:
Mocks of these functions are in fact useless 👍 
Tests are more relevant this way.